### DISCUSSION TITLE
Has anyone tried dynamic serving with Razzle?

### DISCUSSION THREAD

**Discussion by crup** [jaredpalmer/razzle#1388] [Q&A] [answered]
## ❓Question
Dynamic serving: 
https://developers.google.com/search/mobile-sites/mobile-seo/dynamic-serving

Objective: 
To create two bundles - one for mobile and other one for desktop and serve them after evaluating the user agent string on server. 

Has anyone tried serving multiple bundles with Razzle or is there a boilerplate available for it?

**Comment by fivethreeo** [comment, marked as answer]
https://github.com/jaredpalmer/razzle/issues/729

**Comment by fivethreeo** [comment]
There are no examples of this. Also there is no support for multiple builds. This should be added and there currently is no plan for this. Any direct proposals or code to fix this would be appreciated 😃

**Comment by fivethreeo** [comment]
But if you somehow make two builds this is entirely possible 😀

**Comment by kokjinsam** [comment]
I was looking to do similar thing like what OP stated. After a couple times of hacking this and that, I figured that separating mobile and desktop builds is just not worth the effort. So, here's what I did:

1.  Determine screen size using `user-agent` on server side. There's one caveat though, IPhone X has a huge landscape size of 812 pts. So this might cause React to throw hydration mismatch error if your mobile layout doesn't support up to 812px.
2. Pass that screen size as `<Root />` props. Let your Root component render the respective layout.
3. If you're using `loadable-component`, prefix your import magic comment like such: `webpackChunkName: "MobileHome"` or whatever that fits your workflow so that you can filter out chunks that you don't wanna include in your final HTML. 

So far I've not encountered any problems yet :sweat_smile:
